tools/repeatmodeler/main.nf

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env nextflow
+
+// Specify DSL2
+nextflow.enable.dsl=2
+
+// This module uses the Dfam consortium's very handy container.
+// See https://github.com/Dfam-consortium/TETools/ for how to add
+// custom databases to the container.
+
+// Process definition
+process repeatmodeler_database {
+    label "min_cores"
+    label "min_mem"
+    label "regular_queue"
+
+    tag "${meta.sample_id}"
+
+    publishDir "${params.outdir}/${opts.publish_dir}",
+        mode: "copy",
+        overwrite: true,
+        saveAs: { filename ->
+                      if (opts.publish_results == "none") null
+                      else filename }
+
+    container "dfam/tetools:1.2"
+    containerOptions '-u \$(id -u):\$(id -g) -v "$PWD":/work --env "HOME=/work"'
+
+    input:
+        val opts
+        tuple val(meta), path(fasta)
+
+    output:
+        tuple val(meta), path("*{nhr,nin,nnd,nni,nog,nsq,translation}"), emit: repeatmodeler_db
+
+    script:
+
+    args = ""
+    if(opts.args) {
+        ext_args = opts.args
+        args += ext_args.trim()
+    }
+
+    repeatmodeler_database_command = "BuildDatabase $args -name ${fasta.simpleName} ${fasta}"
+
+    if (params.verbose){
+        println ("[MODULE] repeatmodeler_database_command command: " + repeatmodeler_database_command)
+    }
+
+    //SHELL
+    """
+    ${repeatmodeler_database_command}
+    """
+}
+
+process repeatmodeler_model {
+    label "avg_cores"
+    label "avg_mem"
+    label "regular_queue"
+
+    tag "${meta.sample_id}"
+
+    publishDir "${params.outdir}/${opts.publish_dir}",
+        mode: "copy",
+        overwrite: true,
+        saveAs: { filename ->
+                      if (opts.publish_results == "none") null
+                      else filename }
+
+    container "dfam/tetools:1.2"
+    containerOptions '-u \$(id -u):\$(id -g) -v "$PWD":/work --env "HOME=/work"'
+
+    input:
+        val opts
+        tuple val(meta), path(fasta)
+        tuple val(meta), path(repeatmodeler_db)
+
+    output:
+        tuple val(meta), path("*/consensi.fa"), emit: fasta
+        tuple val(meta), path("*/families.stk"), emit: stockholm
+        tuple val(meta), path("*"), emit: repeatmodeler
```

Comment:
Will you use this all channel?

Reply:
The processes create a ton of temporary files with time-stamped names so a wildcard is nice to capture that. 

As well, the majority of the work might proceed but fail to produce the expected `consensi.fa` or `families.stk` outputs. It's expected that repeatmodeler users restart a failed run by manually going in to the temp directory. This doesn't work well within the context of nextflow. I can imagine that a user might not want to start over if the assembly pipeline fails because of repeatmodeler, so it might be useful to retain these temporary files.